clustered/common/src/main/java/org/ehcache/clustered/common/internal/exceptions/ClusteredEhcacheException.java

```diff
@@ -29,4 +30,14 @@ public ClusteredEhcacheException(String message, Throwable cause) {
   public ClusteredEhcacheException(Throwable cause) {
     super(cause);
   }
+
+  /**
+   * Makes a copy of this {@code ClusteredEhcacheException}, retaining the type, in the current
+   * client context.  This method is used to prepare exceptions originating in the server to be
+   * re-thrown in the client.
+   *
+   * @return a new {@code ClusteredEhcacheException} of {@code this} subtype with {@code this}
+   *      instance as the cause
+   */
+  public abstract ClusteredEhcacheException withClientStackTrace();
```

Comment:
Exceptions and Generics can't be mixed == "Error:(6, 42) java: a generic class may not extend java.lang.Throwable"